test/junit/scala/util/RandomTest.scala

```diff
@@ -0,0 +1,15 @@
+package scala.util
+
+import org.junit.{ Assert, Test }
+
+class RandomTest {
+  // Test for SI-9059
+  @Test def testAlphanumeric: Unit = {
+    def isAlphaNum(c: Char) = (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')
+
+    val items = Random.alphanumeric.take(100000)
+    for (c <- items) {
+      Assert.assertTrue(s"$c should be alphanumeric", isAlphaNum(c))
```

Comment:
Footnote, this is a correctness test, not an efficiency test. It's an inefficient correctness test.